tests/lib/rules/no-setup-props-destructure.js

```diff
@@ -504,21 +504,6 @@ tester.run('no-setup-props-destructure', rule, {
         }
       ]
     },
-    {
-      filename: 'test.vue',
-      code: `
-      <script setup>
-      const {count} = defineProps({count:Number})
-      </script>
-      `,
-      errors: [
-        {
-          message:
-            'Destructuring the `props` will cause the value to lose reactivity.',
-          line: 3
-        }
-      ]
-    },
```

Comment:
Should we make this behavior dependent on the Vue version? Or make it configurable to still warn about this? (But not in the deprecated rule.)

Also, why is the same change not necessary in the `no-setup-props-reactivity-loss` tests?

Reply:
> Should we make this behavior dependent on the Vue version? Or make it configurable to still warn about this? (But not in the deprecated rule.)

If necessary, we should create a new rule for Vue2. However, since Vue2 has already reached its EOL, I'm wondering if a new rule are really necessary 🤔 

> Also, why is the same change not necessary in the `no-setup-props-reactivity-loss` tests?

Since `no-setup-props-reactivity-loss` was a deprecated rule and an alias, I wasn't planning on maintaining the test for it.
However, changes to the original rule caused a test to fail, so I removed it.

Reply:
The new test was added to `no-setup-props-reactivity-loss.js` 👍 

https://github.com/vuejs/eslint-plugin-vue/pull/2550/files#diff-0c8677d8ffefa333a1434ec8b0c44ca42be0f2553d6bd7f1ac315368007787e2R540-R544

Reply:
Yes, I've seen that.

I was just wondering why the following test case was not part of `no-setup-props-reactivity-loss` before (because that should have been reported before Vue 3.5). Or if it is included, why it still is not failing (because it should not report this case anymore):

```vue
<script setup>
const {count} = defineProps({count:Number})
</script>
```

Reply:
I understand, thanks for the explanation 😄 
I added that test case as a `valid` test case.